subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/junit/result/JUnitXmlResultOptions.java

```diff
@@ -20,9 +20,13 @@ public class JUnitXmlResultOptions {
 
     public final boolean outputPerTestCase;
     public final boolean mergeReruns;
+    public final boolean omitSystemOutLog;
+    public final boolean omitSystemErrLog;
 
-    public JUnitXmlResultOptions(boolean outputPerTestCase, boolean mergeReruns) {
+    public JUnitXmlResultOptions(boolean outputPerTestCase, boolean mergeReruns, boolean omitSystemOutLog, boolean omitSystemErrLog) {
```

Comment:
🤔 Rather than changing all the places where instances of this type are instantiated, it might be a good idea to add another constructor that forwards to this one using defaults for the new flags which preserve the existing behavior:

```
public JUnitXmlResultOptions(boolean outputPerTestCase, boolean mergeReruns) {
  this(outputPerTestCase, mergeReruns, false, false);
}
```

Reply:
in that case, we have only one usage of the constructor in AbstractTestTask:527 where we use parameters passed from the task. The new constructor won't be used anywhere else.